<?xml version="1.0" ?>

<robot name="servicebot" xmlns:xacro="http://ros.org/wiki/xacro">
<xacro:property name="M_PI" value="3.1415926535897931" />
   <material name="body_color">
    <color rgba="0.6 0.6 0.6 1"/>
  </material>

    <link name="base_footprint"/>

    <joint name="base_joint" type="fixed">
      <origin xyz="0 0 0.08" rpy="0 0 0" />
      <parent link="base_footprint"/>
      <child link="base_link" />
    </joint>

    <link name="base_link">
      <visual>
        <geometry>
          <!-- new mesh -->
          <mesh filename="package://linorobot/urdf/meshes/ServiceBot.stl" scale="0.001 0.001 0.001"/>
        </geometry>
        <material name="body_color" />
        <origin xyz="0 0 -0.08" rpy="0 0 0"/>
      </visual>
    </link>


    <joint name="imu_joint" type="fixed">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="base_link"/>
      <child link="imu_link"/>
    </joint>

    <link name="imu_link">
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </link>

    <joint name="laser_joint" type="fixed">
      <origin xyz="0 0 0.0" rpy="0 0 ${M_PI}"/>
      <parent link="base_link"/>
      <child link="laser"/>
    </joint>

     <link name="laser">
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </link>

    <joint name="head_cam_joint" type="fixed">
      <origin xyz="0.145 0 0.265" rpy="0 ${M_PI/2} 0"/>
      <parent link="base_link"/>
      <child link="head_cam"/>
    </joint>

     <link name="head_cam">
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </link>

    <joint name="ultrasonic_scan_joint" type="fixed">
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <parent link="base_link"/>
      <child link="ultrasonic_link"/>
    </joint>

    <link name="ultrasonic_link">
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </link>

</robot>
